pkg/manager/manager.go

```diff
@@ -339,7 +339,13 @@ func New(config *rest.Config, options Options) (Manager, error) {
 	if leaderConfig == nil {
 		leaderConfig = rest.CopyConfig(config)
 	}
-	resourceLock, err := options.newResourceLock(leaderConfig, recorderProvider, leaderelection.Options{
+
+	leaderRecorderProvider, err := options.newRecorderProvider(leaderConfig, cluster.GetScheme(), options.Logger.WithName("events"), options.makeBroadcaster)
```

Comment:
Can we new this `leaderRecorderProvider`  only if `options.LeaderElectionConfig` is not nil? Otherwise it should be redundant to `recorderProvider`.